### PR TITLE
Docs: Update CONTRIBUTING.md with shortcut command for assembling only the tar distribution

### PR DESCRIPTION
**Pull request by cjcenizal**
This will help people assemble distributions without running into a problem I ran into when running the normal `assemble` command, in which `buildBwcVersion` failed because I lacked old Java versions:

```sh
> Task :distribution:bwc:bugfix:buildBwcVersion FAILED

 [6.4.3] ERROR: JAVA_HOME is set to an invalid directory: null
 [6.4.3]
 [6.4.3] Please set the JAVA_HOME variable in your environment to match the
 [6.4.3] location of your Java installation.
```

### CODE REVIEW

File: CONTRIBUTING.md
Comment (jasontedor) [inline CONTRIBUTING.md]: `all distributable artifacts` -> `all build artifacts`

File: CONTRIBUTING.md
Comment (jasontedor) [inline CONTRIBUTING.md]: `distributable` -> `build`
`(e.g. plugins, javadocs)` -> `(e.g., plugins and Javadocs)`
`javadocs` -> `Javadocs`

**Review by jasontedor — COMMENTED** [review]
I left a few nits, I'll defer to @nik9000 for overall approval.

**Comment by cjcenizal** [comment]
Thanks @jasontedor and @nik9000! I've addressed your feedback. Could you take another look? I've also added a note about the `--parallel` flag.

File: CONTRIBUTING.md
Comment (jasontedor) [reply CONTRIBUTING.md]: It does make the build process faster, since pieces of compilation can occur as the dependencies are ready.

Comment (jasontedor) [reply CONTRIBUTING.md]: And it applies to both.

Comment (cjcenizal) [reply CONTRIBUTING.md]: Great, thanks!

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by cjcenizal** [comment]
@jasontedor Do I need to get this PR to pass CI?

@nik9000 I'll merge once I get your approval.

**Comment by nik9000** [comment]
Lgtm! Thanks for working on it!

**Comment by nik9000** [comment]
And CI doesn't look at this file so I wouldn't worry about it.

**Comment by jasontedor** [comment]
>Should I backport this to 6.x?

I expect most folks don't look at contributing in the non-master branch, but I do like when we keep our development branches up to date. So I would be good with cherry-picking this back to 6.x.

**Pull request merged**
